presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -56,7 +56,7 @@ private static DistributionType determineDistributionType(JoinNode node, Context
             return PARTITIONED;
         }
 
-        if (node.getCriteria().isEmpty() && type == INNER) {
```

Comment:
add this change also to legacy optimizer

Reply:
This i didn't want to do, so that there is a switch allowing to disable the rule.